security/tokens_and_compartments.rst

```diff
@@ -0,0 +1,212 @@
+
+Patient-level scopes
+--------------------
+For Patient-level scopes the access control engine in Firely Server evaluates two types of authorization:
+
+#. Type-Access: Is the user allowed to read or write resource(s) of a specific resourcetype?
+#. Compartment: Is the data to be read or written within the current compartment (if any)?
+
+As you may have noticed, Type-Access aligns with the concept of scopes, and Compartment aligns with the concept of launch context in SMART on FHIR.
+Scopes are evaluated by checking if the application has been granted the correct rights to execute a certain CRUD operation.
+The compartment access is evaluated by the filters defined in the SMART appsettings. See :ref:`feature_accesscontrol_config`::
+
+   "Filters": [
+      {
+        "FilterType": "Patient", //Filter on a Patient compartment if a 'patient' launch scope is in the auth token
+        "FilterArgument": "_id=#patient#" //... for the Patient that has an id matching the value of that 'patient' launch scope
+      },
+      ...
+   ]
+
+For example, the authorization server provides a patient launch context with the value `123`. This is to be interpreted by the application as "the user instructs the application to work in the context of the resource `Patient/123`".
+Firely Server internally forms a compartment around all resources that are linked to `Patient/123` according to the Patient CompartmentDefinition.
+
+.. note::
+  To enable access to resources outside the compartment, the client must request additional scopes for these resources specifically.
+
+There may be cases where the logical id of the focus resource is not known to the authorization server. Let's assume it does know one of the identifiers of a Patient. The Filters in the :ref:`feature_accesscontrol_config` allow you to configure Firely Server to use the identifier search parameter as a filter instead of _id::
+
+   "Filters": [
+      {
+        "FilterType": "Patient", //Filter on a Patient compartment if a 'patient' launch scope is in the auth token
+        "FilterArgument": "identifier=#patient#" //... for the Patient that has an id matching the value of that 'patient' launch scope
+      },
+      ...
+   ]
+
+Please notice that it is possible that more than one Patient matches the filter. This is intended behaviour of Firely Server, and it is up to you to configure a search parameter that is guaranteed to have unique values for each Patient if you need that.
+However you can also take advantage of it and allow access only to the patients from a certain General Practitioner, of whom you happen to know the Identifier::
+
+   "Filters": [
+      {
+        "FilterType": "Patient", //Filter on a Patient compartment if a 'patient' launch scope is in the auth token
+        "FilterArgument": "general-practitioner.identifier=#patient#" //... for the Patient that has an identifier matching the value of that 'patient' launch scope
```

Comment:
In the comment, do you mean "... for the Practitioner that has an identifier ..."?